app/javascript/src/utils/compiler.js

```diff
@@ -410,3 +411,101 @@ function evaluateForLoops(joinedItems) {
+    // because the parenthesis group should be taken as one value (e.g. for the previous
+    // example, we should return ["1", "(2, 3)", "4"], not ["1", "(2", "3)", "4"])
+    const openBracketIndex = input.indexOf("(", nextStartingIndex)
+    if (openBracketIndex >= 0 && openBracketIndex < commaMatch.index) {
+      const closingBracketIndex = getClosingBracket(input, "(", ")", openBracketIndex - 1)
+      nextStartingIndex = closingBracketIndex < 0 ? input.length : closingBracketIndex
+    } else {
+      const commaEndIndex = commaMatch.index + commaMatch[0].length - 1
+      result.push(input.substring(lastValidCommaEndIndex + 1, commaMatch.index))
+
+      lastValidCommaEndIndex = commaEndIndex
+      nextStartingIndex = lastValidCommaEndIndex + 1
+    }
+
+    commaRegex.lastIndex = nextStartingIndex
+  }
+
+  const lastValue = input.substring(lastValidCommaEndIndex + 1)
+  if (lastValue.length > 0) {
+    result.push(lastValue)
+  }
+  return result
+}
+
+function evaluateEachLoops(joinedItems) {
+  const eachRegex = /@each\s*\((\w+)(?:,\s+(\w+))?\s+in\s+((?:.|\n)+?)\s*\)\s*\{/g
+
+  let match
+  while ((match = eachRegex.exec(joinedItems)) != null) {
+    const [_, valueVar, indexVar, iterableStr] = match
+
+    let iterable = []
+    if (iterableStr[0] === "[" && iterableStr[iterableStr.length - 1] === "]") {
+      iterable = parseArrayValues(iterableStr.substring(1, iterableStr.length - 1))
+    } else {
+      const language = get(defaultLanguage)
+      const constants = get(workshopConstants)
+
+      const sanitizedIterableStr = iterableStr.toLowerCase().trim()
+
+      // TODO: support constant names in other languages
+
+      const usedConstantName = Object.keys(constants)
+        .find((name) => {
+          const sanitizedName = name.toLowerCase()
+          return sanitizedName === sanitizedIterableStr ||
+            `${ sanitizedName }s` === sanitizedIterableStr // naive way to allow plurals, which is more natural (e.g. "value in Buttons")
+        })
+
+      if (usedConstantName != null) {
+        iterable = Object.values(constants[usedConstantName]).map((value) => value[language])
+      }
+    }
+
+    if (iterable == null) {
+      continue
+    }
+
+    const openingBracketIndex = match.index + match[0].length - 1
+    const closingBracketIndex = getClosingBracket(joinedItems, "{", "}", openingBracketIndex - 1)
+    if (closingBracketIndex < 0) {
+      continue
+    }
+
+    const contentToRepeat = joinedItems.substring(openingBracketIndex + 1, closingBracketIndex)
+    const indexVarRegex = new RegExp(`Each.${ indexVar || "i" }(?=\\W|$)`, "g")
+    const valueVarRegex = new RegExp(`Each.${ valueVar }(?=\\W|$)`, "g")
+
+    let finalContent = ""
+    // Replace "Each.[valueVar]" and "Each.[indexVar]"
+    for (const [index, value] of Object.entries(iterable)) {
+      finalContent += contentToRepeat
+        .replaceAll(indexVarRegex, index)
+        .replaceAll(valueVarRegex, value)
+    }
```

Comment:
To be a little more consistent it might be better to use .forEach or .reduce, `for...of` isn't used anywhere I believe.

Something like this might be nice enough
```suggestion
const finalContent = Object.entries(iterable).reduce((current, [index, value]) => {
  return current + contentToRepeat
    .replaceAll(indexVarRegex, index)
    .replaceAll(valueVarRegex, value)
}, "")
```
Or otherwise a regular for loop.